internal/experimental/registry/client.go

```diff
@@ -207,6 +214,11 @@ type (
 
 // Pull downloads a chart from a registry
 func (c *Client) Pull(ref string, options ...PullOption) (*PullResult, error) {
+	// Convert plus (+) to underscore (_) in request string before pull
+	// This must remain right at the top of client.Pull()
+	// See https://github.com/helm/helm/issues/10166
+	ref = strings.ReplaceAll(ref, "+", "_")
```

Comment:
I'm sorry I didn't catch this earlier. The `+` should only be changed in the tag and not the whole `ref`. A `ref` is a URI and other parts of the URI (e.g., the path) can contain a `+` character. It's not as easy as replacing all the + with _.

Reply:
Note, this applies to all places there the swap happens.

Reply:
Ah yes, good catch. Myself and several other people are divvying up tasks for wrapping this up, so either I or someone else will push a fix to this ASAP ✨

Reply:
leaving an update on the approach (pushed in fb0eeeb) here after talking with @scottrigby:

to scope to a tag we first need to identify a tag, and to identify a tag we _should_ first validate the reference.  so fb0eeeb first identifies a tag and does the `'+' -> '_'` replacement only on the tag if it exists, then returns a validated reference (using oras' `registry.ParseReference`) or error.  the remainder of `Push`/`Pull` then uses the validated `oras.Reference` struct for all ops.

while this means adding the updated oras dependency to this PR, it will eventually be consolidated to the updated oras release alongside #10527